docs/QuickStart.md

```diff
@@ -0,0 +1,204 @@
+
+## Start a Client
+Client first creates a Spider client driver and connects it to the database. Spider automatically cleans up the resource in driver's destructor, but you can close the driver to release the resource early.
+```c++
+#include <spider/Spider.hpp>
+
+auto main(int argc, char **argv) -> int {
+    spider::Driver driver{};
+    driver.connect("db_url");
+    
+    driver.close();
+}
+```
+
+## Create a Task
+In Spider, a task is a non-member function that takes the first argument a `spider::Context` object. It can then take any number of arguments of POD type.
+
+Task can return any POD type. If a task needs to return more than one result, uses `std::tuple`.
+
+The `Context` object represents the context of a running task. It provides methods to get the task metadata information like task id. It also supports the creating task inside a task. We will cover this later.
+```c++
+auto sum(spider::Context &context, int x, int y) -> int {
+    return x + y;
+}
+
+auto sort(spider::Context &context, int x, int y) -> std::tuple<int, int> {
+    if (x >= y) {
+        return { x, y };
+    }
+    return { y, x };
+}
+```
+
+## Run a Task
+Spider enables user to run a task on the cluster. First register the functions statically so it is known by Spider. Simply call `Driver::run` and provide the arguments of the task. `Driver::run` returns a `spider::Future` object, which represents the result that will be available in the future. You can call `Future::ready` to check if the value in future is available yet. You can use `Future::get` to block and get the value once it is available. 
+```c++
+spider::register_task(sum);
+spider::register_task(sort);
+
+auto main(int argc, char **argv) -> int {
+    // driver initialization skipped
+    spider::Future<int> sum_future = driver.run(sum, 2);
+    assert(4 == sum_future.get());
+
+    spider::Future<std::tuple<int, int>> sort_future = driver.run(4, 3);
+    assert(std::tuple{3, 4} == sort_future.get());
+}
+```
+
+## Group Tasks Together
+In real world, running a single task is too simple to be useful. Spider lets you bind outputs of tasks as inputs of another task, similar to `std::bind`. Binding the  tasks together forms a dependencies among tasks, which is represented by `spider::TaskGraph`. `TaskGraph` can be further bound into more complicated `TaskGraph` by serving as inputs for another task. You can run the task using `Driver::run` in the same way as running a single task.
+```c++
+auto square(spider::Context& context, int x) -> int {
+    return x * x;
+}
+
+auto square_root(spider::Context& context, int x) -> int {
+    return sqrt(x);
+}
+// task registration skipped
+auto main(int argc, char **argv) -> auto {
+    // driver initialization skipped
+    spider::TaskGraph<int(int, int)> sum_of_square = spider::bind(sum, square, square);
+    spider::TaskGraph<int(int, int)> rss = spider::bind(square_root, sum_of_square);
+    spider::Future<int> future = driver::run(rss, 3, 4);
```

Comment:
I don't really understand how arguments are passed between tasks. You mention that this interface is similar to `std::bind`, but if I were to replace every call to `spider::bind` with `std::bind`, I don't think this example would work. So from both a design perspective and a usage perspective, we need to clear this up.

Reply:
I've updated the doc with more explanation, but I don't think my description is clear enough. I need some advice on this.